src/components/checkBoxSelection/checkBoxSelection.js

```diff
@@ -1,47 +1,108 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import './checkBoxSelection.scss';
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "./checkBoxSelection.scss";
+import { connect } from "react-redux";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class CheckBoxSelection extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: false
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected
+    };
+  }
+
+  convertSvgToDataUrl = file => {
+    let reader = new FileReader();
+    reader.onload = result => {
+      this.props.reduxState.svgSettingList.push({
+        name: file.name,
+        dataurl: result.target.result
+      });
+    };
+    if (file) {
+      reader.readAsDataURL(file);
     }
+  };
 
-    componentWillMount() {
-	 this.setState({ isClicked: this.props.isSelected }); 
+  changeColourAndTick = () => {
+    let isFound = false;
+    let index = 0;
+    this.setState({ isClicked: !this.state.isClicked });
+    for (var key in this.props.reduxState.svgSettingList) {
+      if (
+        this.props.reduxState.svgSettingList[key].name === this.props.file.name
+      ) {
+        isFound = true;
+        index = key;
+      }
```

Comment:
.

Reply:
I do not understand what do you mean by a "." (full stop)

Reply:
the full stop was sent before you published your review message

Reply:
okay, so do you still consider me to change this ?

Reply:
yes please, do you disagree that it makes sense?

Reply:
Yes, I disagree because I have to find not only an item but its index too and using two loops the one in  `find` and in `indexOf` while I can loop once and get both my item and its index. unless find can have an argument to keep track of index. I will have to look to that and get back to your request maybe its makes sense to use find.